GenderPayGap.UnitTests/GenderPayGap.WebUI.Tests/CodeQualityTests/DoNotUseRedirectWithReturnUrls.cs

```diff
@@ -45,19 +46,22 @@ public void DoNotAllowRedirectWithReturnUrlInCode()
                     continue;
                 }
 
-                string fileText = File.ReadAllText(filePath);
-                // There are a couple of actions on controllers Redirect() that wrap a RedirectToAction,
-                // we want to exclude these from the check as we are only concerned about redirecting with a url parameter
-                if (fileText.Contains(" Redirect(") && !fileText.Contains(" Redirect()"))
+                // We want to pick up all instances of Redirect(url) or new RedirectResult(url) that do not have the disable comment on the previous line.
+                // Ignores those without parameters so we don't pick up any methods defined as Redirect() for example
+                var lines = File.ReadAllLines(filePath);
+                for (var i = 0; i<lines.Length; i++)
                 {
-                    failedFiles.Add(filePathSuffix);
+                    if (Regex.IsMatch(lines[i], @"(new){0,1}\s+Redirect(Result){0,1}\s*\(\S+\)") && !lines[i-1].Contains("//disable:DoNotUseRedirectWithReturnUrls"))
```

Comment:
I think it's generally better/faster to use a compiled regex here - https://docs.microsoft.com/en-us/dotnet/standard/base-types/best-practices so you'd pull out a `var redirectResult = new Regex(....)` outside of the loop/test method and use that